pkg/client/issuecredential/client.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package issuecredential
+
+import (
+	"errors"
+
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/common/service"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/issuecredential"
+)
+
+// Provider contains dependencies for the issuecredential protocol and is typically created by using aries.Context()
+type Provider interface {
+	Service(id string) (interface{}, error)
+}
+
+// ProtocolService defines the issuecredential service.
+type ProtocolService interface {
+	service.DIDComm
+	Actions() ([]issuecredential.Action, error)
+	ActionContinue(piID string, opt issuecredential.Opt) error
+	ActionStop(piID string, err error) error
+}
+
+// Client enable access to issuecredential API
+type Client struct {
+	service.Event
+	service ProtocolService
+}
+
+// New return new instance of the issuecredential client
+func New(ctx Provider) (*Client, error) {
+	raw, err := ctx.Service(issuecredential.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	svc, ok := raw.(ProtocolService)
+	if !ok {
+		return nil, errors.New("cast service to issuecredential service failed")
+	}
+
+	return &Client{
+		Event:   svc,
+		service: svc,
+	}, nil
+}
+
+// Actions returns unfinished actions for the async usage
+func (c *Client) Actions() ([]issuecredential.Action, error) {
+	return c.service.Actions()
+}
+
+// SendOffer is used by the Issuer to send an offer.
+func (c *Client) SendOffer(offer *issuecredential.OfferCredential, myDID, theirDID string) error {
```

Comment:
I feel having a target connection would make more sense than passing myDID and theirDID (internally service/client can derive this). 

ie, Issue credential to target connectionID - SendOffer(offer *issuecredential.OfferCredential, connectionID string)

Reply:
It is not a problem. It can be replaced at any time. Anyway, the connection should contain myDID and theirDID.